transformer_engine/pytorch/module/base.py

```diff
@@ -87,9 +87,140 @@ def initialize_ub(
+                     "cga_size": 2,
+                     "set_sm_margin": True,
+                     "fp8_buf": True,
+                 },
+                 "proj_fprop": {
+                     "method": "pipeline",
+                     "is_reduce_scatter": True,
+                     "num_sm": 16,
+                     "cga_size": 2,
+                     "set_sm_margin": True,
+                     "num_splits": 4,
+                     "atomic_gemm": False,
+                     "fp8_buf": False,
+                 },
+                 "proj_dgrad": {
+                     "method": "ring_exchange",
+                     "is_reduce_scatter": False,
+                     "num_sm": 1,
+                     "cga_size": 1,
+                     "set_sm_margin": False,
+                     "num_splits": tp_size,
+                     "aggregate": False,
+                     "atomic_gemm": False,
+                     "use_ce": True,
+                     "fp8_buf": True,
+                 },
+                 "fc1_fprop": {
+                     "method": "ring_exchange",
+                     "is_reduce_scatter": False,
+                     "num_sm": 1,
+                     "cga_size": 1,
+                     "set_sm_margin": False,
+                     "num_splits": tp_size,
+                     "aggregate": False,
+                     "atomic_gemm": False,
+                     "use_ce": True,
+                     "fp8_buf": True,
+                 },
+                 "fc1_dgrad": {
+                     "method": "bulk",
+                     "is_reduce_scatter": False,
+                     "num_sm": 16,
+                     "cga_size": 2,
+                     "set_sm_margin": True,
+                     "fp8_buf": True,
+                 },
+                 "fc1_wgrad": {
+                     "method": "bulk",
+                     "is_reduce_scatter": True,
+                     "num_sm": 16,
+                     "cga_size": 2,
+                     "set_sm_margin": True,
+                     "fp8_buf": False,
+                 },
+                 "fc2_fprop": {
+                     "method": "pipeline",
+                     "is_reduce_scatter": True,
+                     "num_sm": 16,
+                     "cga_size": 2,
+                     "set_sm_margin": True,
+                     "num_splits": 4,
+                     "atomic_gemm": False,
+                     "fp8_buf": False,
+                 },
+                 "fc2_dgrad": {
+                     "method": "ring_exchange",
+                     "is_reduce_scatter": False,
+                     "num_sm": 1,
+                     "cga_size": 1,
+                     "set_sm_margin": False,
+                     "num_splits": tp_size,
+                     "aggregate": False,
+                     "atomic_gemm": False,
+                     "use_ce": True,
+                     "fp8_buf": True,
+                 },
+             }
```

Comment:
There is too much detailed information and it's very likely to get out of sync:
```suggestion
    ub_cfgs: dict, optional
             Configuration dictionary containing a Userbuffers options for each GEMM layer in a
             te.TransformerLayer.
```

Reply:
I'll condense the documentation here but we've gotten several requests to make this config information available in the docstring for this initializer, so I think disclosing at least the skeleton structure + layer names would be valuable.